src/huggingface_hub/commands/repo.py

```diff
@@ -0,0 +1,310 @@
+                exit()
+        try:
+            url = self._api.create_repo(
+                repo_id=repo_id,
+                token=self.token,
+                repo_type=self.args.type,
+                space_sdk=self.args.space_sdk,
+            )
+        except HTTPError as e:
+            print(e)
+            print(ANSI.red(e.response.text))
+            exit(1)
+        print("\nYour repo now lives at:")
+        print(f"  {ANSI.bold(url)}")
+        print("\nYou can clone it locally with the command below, and commit/push as usual.")
+        print(f"\n  git clone {url}")
+        print("")
+
+
+class RepoListCommand(BaseRepoCommand):
+    def run(self):
+        self.type = self.args.type
+        self.author = self.args.author
+        self.search = self.args.search
+        try:
+            if self.type is None or self.type == "model":
+                repos = self._api.list_models(token=self.token, author=self.author, search=self.search)
+            elif self.type == "dataset":
+                repos = self._api.list_datasets(token=self.token, author=self.author, search=self.search)
+            elif self.type == "space":
+                repos = self._api.list_spaces(token=self.token, author=self.author, search=self.search)
+        except HTTPError as e:
+            print(e)
+            print(ANSI.red(e.response.text))
+            exit(1)
+        print("\nYour repos:")
+        for repo in repos:
+            print(f"  {ANSI.bold(repo.id)}")
+        print("")
+
+
+class RepoDeleteCommand(BaseRepoCommand):
+    def run(self):
+        user = self._api.whoami(self.token)["name"]
+        namespace = self.args.organization if self.args.organization is not None else user
+
+        repo_id = f"{namespace}/{self.args.name}"
+
+        if self.args.type not in REPO_TYPES:
+            print("Invalid repo --type")
+            exit(1)
+
+        if self.args.type in REPO_TYPES_URL_PREFIXES:
+            prefixed_repo_id = REPO_TYPES_URL_PREFIXES[self.args.type] + repo_id
+        else:
+            prefixed_repo_id = repo_id
+
+        print(f"You are about to delete {ANSI.bold(prefixed_repo_id)}")
+
+        if not self.args.yes:
+            choice = input("Proceed? [Y/n] ").lower()
+            if not (choice == "" or choice == "y" or choice == "yes"):
+                print("Abort")
+                exit()
+        try:
+            self._api.delete_repo(repo_id=repo_id, token=self.token, repo_type=self.args.type)
+        except HTTPError as e:
+            print(e)
+            print(ANSI.red(e.response.text))
+            exit(1)
+        print("\nYour repo has been deleted.")
+        print("")
+
+
+class RepoToggleCommand(BaseRepoCommand):
+    def run(self):
+        self.private = self.args.private
+        user = self._api.whoami(self.token)["name"]
+        namespace = self.args.organization if self.args.organization is not None else user
+
+        repo_id = f"{namespace}/{self.args.name}"
```

Comment:
For once, doing a `whoami` to get the namespace (if not provided) by the user will be really useful as it cannot be implicitly used (in opposition to create_repo/delete_repo)